File: frontend/src/types/market.ts
export interface CommodityPrice {
  name: string;
  ticker: string;
  current_price: number;
  currency: string;
  unit?: string;
  change_30d: number | null;
}

export interface HistoricalPrice {
  date: string;
  [commodity: string]: string | number;
}

export interface InterestRate {
  value: number | null;
  label: string;
  series_id?: string;
  as_of?: string;
  error?: string;
}

export interface NewsItem {
  title: string;
  link: string;
  published: string;
  source: string;
  summary: string;
}

export interface MarketSummary {
  summary_text: string;
  commodities_used: number;
  news_count: number;
}

File: backend/app/models/market.py
from pydantic import BaseModel
from typing import Optional


class CommodityPrice(BaseModel):
    name: str
    ticker: str
    current_price: float
    currency: str = "USD"
    change_30d: Optional[float] = None


class InterestRate(BaseModel):
    name: str
    value: Optional[float]
    series_id: str
    as_of: Optional[str] = None


class NewsItem(BaseModel):
    title: str
    link: str
    published: str
    source: str
    summary: str = ""
